packages/api/src/resolvers/mutations/index.ts

```diff
@@ -155,13 +155,19 @@ import prepareProductMediaUpload from './products/prepareProductMediaUpload.js';
 import prepareAssortmentMediaUpload from './assortments/prepareAssortmentMediaUpload.js';
 import prepareUserAvatarUpload from './users/prepareUserAvatarUpload.js';
 import rejectOrder from './orders/rejectOrder.js';
+import removePushNotificationSubscription from './users/removePushNotificationSubscription.js';
+import savePushNotificationSubscription from './users/savePushNotificationSubscription.js';
+import pushNotificationEnabled from './users/pushNotificationEnabled.js';
 
 export default {
   logout,
   logoutAllSessions,
   loginAsGuest,
   createWebAuthnCredentialCreationOptions,
   createWebAuthnCredentialRequestOptions,
+  pushNotificationEnabled: acl(actions.updateUser)(pushNotificationEnabled),
```

Comment:
And, actually, the server cannot know if push notifications are enabled because you can mute notifications client-side. The server can only know if there is registered subscriptions of user agents.

Rather than the above, why not add something like:

User.pushSubscriptions: [PushSubscription!]!

Reply:
@pozylon the boolean flag comes necessary when a user has enabled push notifications on their browser for other sites but not specifically on an unchained engine. 
I don't see any issue with attaching all of the push subscription objects to a user but I think we also need to include that flag to lift the task of checking if a user has subscribed to a specific browser from the client side.

Reply:
You're refering to this logic here: https://github.com/unchainedshop/unchained/pull/544/files#diff-3e3712661a9e4151bd1be7b312bf5c143e994d20193dd9ddf3262fc927796693

So that logic is anyways wrong: Let's assume your user agent string is "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.2 Safari/605.1.15"

How can you find out if the device you're currently using is registered? You can't because i could have another macbook.

The ONLY unique key for a user agent is not the user-agent http header string, it's the "keys" of the subscription object.

Reply:
The push implementation has to make sure that any user at any time can register multiple browsers on multiple devices.

Reply:
The flow should be client-side:

1. Some triggger
2. Get Push Subscription from PushManager (eventually allow permission first)
3. Check if Push Subscription is already registered on the User by comparing the "_id" of User.pushSubscriptions array and the "keys" of the subscription on the client. If not on server, send to server by calling addPushSubscription. You should not remove any subscriptions on other users.

Reply:
Some remark: Theoretically you might want to forbid having subscriptions sent to the browser for users that you have logged in with before. For that case, "addPushSubscription" could take an argument like "unsubscribeFromOtherUsers" which searches for the push subscription key in other users and removes that subscription.